FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Mutual_Auth_wolfSSL/DemoTasks/MutualAuthMQTTExample.c

```diff
@@ -434,19 +434,26 @@ static void prvMQTTDemoTask( void * pvParameters )
     }
 }
 /*-----------------------------------------------------------*/
-
 static void prvTLSConnect( NetworkCredentials_t * pxNetworkCredentials,
                            NetworkContext_t * pxNetworkContext )
 {
     BaseType_t xNetworkStatus;
 
     /* Set the credentials for establishing a TLS connection. */
+
     pxNetworkCredentials->pRootCa = ( const unsigned char * ) democonfigROOT_CA_PEM;
-    pxNetworkCredentials->rootCaSize = sizeof( democonfigROOT_CA_PEM );
     pxNetworkCredentials->pClientCert = ( const unsigned char * ) democonfigCLIENT_CERTIFICATE_PEM;
-    pxNetworkCredentials->clientCertSize = sizeof( democonfigCLIENT_CERTIFICATE_PEM );
     pxNetworkCredentials->pPrivateKey = ( const unsigned char * ) democonfigCLIENT_PRIVATE_KEY_PEM;
-    pxNetworkCredentials->privateKeySize = sizeof( democonfigCLIENT_PRIVATE_KEY_PEM );
+
+    #if defined( democonfigCREDENTIALS_IN_BUFFER )
+        pxNetworkCredentials->rootCaSize = strlen( democonfigROOT_CA_PEM );
```

Comment:
Thank you for your suggestion. Sure, the method you've suggested doesn't require you to have a special file, but instead it brings the data into an existing file that contains the logic or data other than credentials. Do engineers not want to separate data from logic?

Reply:
I am suggesting that `democonfigROOT_CA_PEM` is always defined in `demo_config.h`. If `democonfigCREDENTIALS_IN_BUFFER` is defined to `1` , then `democonfigROOT_CA_PEM` is defined to the certificate data, otherwise it is defined to certificate file path. 

Example configuration if you want to load from buffer:

```c
#define democonfigCREDENTIALS_IN_BUFFER 1
#define democonfigROOT_CA_PEM      \
"-----BEGIN CERTIFICATE-----\n"    \
"...base64 data...\n"              \
"-----END CERTIFICATE-----\n"
```

Example configuration if you want to load from file:

```c
#define democonfigCREDENTIALS_IN_BUFFER 0
#define democonfigROOT_CA_PEM      "Path to RootCA file"
```

Doing so would simplify that instructions in readme file. What do you think?

Reply:
I felt uncomfortable that the data was embedded in demo_config.h along with other settings. However, the method you suggested makes the instructions easier, and it's clear that the user doesn't even have to prepare a file. I will change according to your suggestions.

Reply:
Changed readme.